components/public-api-server/pkg/server/logs.go

```diff
@@ -14,42 +15,38 @@ import (
 func NewLogInterceptor(entry *logrus.Entry) connect.UnaryInterceptorFunc {
 	interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
 		return connect.UnaryFunc(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
-			logger := entry.
-				WithContext(ctx).
-				WithField("protocol", "connect").
-				WithField("procedure", req.Spec().Procedure).
-				WithField("address", req.Peer().Addr).
-				WithField("stream_type", streamType(req.Spec().StreamType))
+			ctx = log.ToContext(ctx, entry.WithContext(ctx))
 
-			isClient := req.Spec().IsClient
-
-			if isClient {
-				logger.WithField("headers", req.Header()).Debugf("Starting request for %s", req.Spec().Procedure)
-			} else {
-				logger.WithField("headers", req.Header()).Debugf("Handling request for %s", req.Spec().Procedure)
-			}
+			log.AddFields(ctx, logrus.Fields{
+				"request.protocol":    "connect",
```

Comment:
Similarly, it would be great to have a "standardized" way to log these things across the app.